recipes/_start.rb

```diff
@@ -10,6 +10,16 @@
   not_if { platform_family?('windows') }
 end
 
+ruby_block 'insert ulimit' do
+  block do
+    file = Chef::Util::FileEdit.new('/etc/init.d/splunk')
+    file.insert_line_after_match(/^RETVAL=\d$/, "ulimit -n #{node['splunk']['limits']['open_files']}")
```

Comment:
The problem is we can't simply prepend it since the line has to come after the magic bash comment that is the first line of the script.